### PR TITLE
Run ethportal-peertest in CI

### PR DESCRIPTION
**Pull request by carver**
Ok, this ended up coalescing into a single PR somewhat reasonably. Basically:

- Move the trin launch into a lib, to make it easy to programmatically launch trin (in this case, for testing)
- Add an easy way to trigger shutdown of trin, rather than just a sys.exit, by shutting down the json-rpc server
- Run ethportal-peertest against a launched trin from within a rust test (instead of via CLI)
- Fixed a hairy bug where it was possible to get a panic during a panic (trying to remove a file that's missing during a panic hook). Debugging that is not fun, definitely something to watch for)
- Added some bonus logs
- Dropped two ping tests, temporarily (See #196)
- Added windows support for peertest

### CODE REVIEW

File: trin-core/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,3 @@
-#[macro_use]
-extern crate lazy_static;
```
Comment (ogenev) [inline trin-core/src/lib.rs:2]: If this is the only use, I guess we need to remove the `lazy_static` dependency from `trin-core/Cargo.toml`

**Review by ogenev — APPROVED** [review]
Nice, looks like a good foundation for future `ethportal-peertest` tests! :+1:

File: tests/self_peertest.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#[cfg(test)]
+mod test {
+    use ethportal_peertest as peertest;
+    use trin_core::cli::TrinConfig;
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_launches() {
+        tracing_subscriber::fmt::init();
+
+        let peertest_config = peertest::PeertestConfig::new_from([".", "--target-node", "enr:-IS4QFuwD0Zgt_R_yRbIMHHQq_iAmewBewahQgIjIiSTRnzZWp_y8atv0jSN0oeE3KJ8qnNHI-AEgfJfqWYVEqvjWqYBgmlkgnY0gmlwhMCoAcCJc2VjcDI1NmsxoQI19BFyTJb6NhSYOi3nJQJdXL3WESUI4ouxYzBr423MdIN1ZHCCIyg"].iter()).unwrap();
```
[the remark this answers is not in the document]
Comment (carver) [reply tests/self_peertest.rs:10]: There's only one client running in the current test, so there's no one to ping. Some discussion about resolving this in #196

File: trin-core/src/jsonrpc/service.rs [hunk 3]
```diff
@@ -165,6 +226,8 @@ fn serve_ipc_client(
 ) {
     let deser = serde_json::Deserializer::from_reader(rx);
     for obj in deser.into_iter::<JsonRequest>() {
+        let rpc_req_str = format!("{:?}", obj);
+        debug!("Got new IPC request: {}", rpc_req_str);
```
[the remark this answers is not in the document]
Comment (carver) [reply trin-core/src/jsonrpc/service.rs:230]: Hah, thanks for catching this. It was cruft from when I had another log at the end of the method, like "IPC request fully handled." At that point, obj had already been consumed.

**Pull request merged**